xds/src/test/java/io/grpc/xds/ClusterResolverLoadBalancerTest.java

```diff
@@ -371,6 +379,109 @@ public void edsClustersEndpointHostname_addedToAddressAttribute() {
             .get(XdsAttributes.ATTR_ADDRESS_NAME)).isEqualTo("hostname1");
   }
 
+  @Test
+  @SuppressWarnings("AddressSelection")
+  public void endpointAddressRewritten_whenProxyMetadataIsInEndpointMetadata() {
+    ClusterResolverConfig config = new ClusterResolverConfig(
+        Collections.singletonList(edsDiscoveryMechanismWithOutlierDetection), leastRequest);
+    deliverLbConfig(config);
+    assertThat(xdsClient.watchers.keySet()).containsExactly(EDS_SERVICE_NAME1);
+    assertThat(childBalancers).isEmpty();
+
+    // Create a fake address for the endpoint
+    EquivalentAddressGroup endpoint =
+        new EquivalentAddressGroup(new InetSocketAddress("endpoint-addr-1", 80));
```

Comment:
This is going to do a DNS resolution and then _hopefully_ create an unresolved InetSocketAddress. It is clearer to do `InetSocketAddress.createUnresolved("endpoint-addr-1" ,80)` (which would also remove the need for `@SuppressWarnings("AddressSelection")`), but it is also weird to use unresolved addresses here. Use something like "127.0.0.1" and "127.0.0.2" for endpoint 1 and endpoint 2? You could use any IP address, really, or the same address and different ports. Let's not use unresolved addresses.